mantidproject/mantid
# Extract TestHelpers into own test target

**Description of work.**
Extracts TestHelpers into it's own test target for a number of reasons:

- Avoid redundant compilation
- Automated tools (clang-tidy) will parse these files once instead of n
times
- Helps MSVC find the various targets
- Makes dep chain explicit.

This also includes some minor modernisation to the CMakelists
associated, such as replacing include_directories with
target_include_directories.

Long term we need to do the following:
- Split out Qt and Framework only helpers
- Consolidate as much implementation into the associated *.cpp files
- Move target specific code (e.g. Muon) into their associated targets


**To test:**
- Check CI builds, this will cause compilation errors if it's incorrect

There is no associated issue. - Part of the clang tidy work

*This does not require release notes* because it's an internal change that has no visible side-effects (apart from faster compilation)



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
